### DISCUSSION TITLE
Format with function call in ui.table

### DISCUSSION THREAD

**Discussion by laserir** [zauberzeug/nicegui#1623] [Q&A] [answered]
### Question

Is it possible to format the ui.table rows using functions, like?:
`{'name': 'level', 'label': 'Level', 'field': 'level', ':format:': 'value => level_mapping(value)'},`
level_mapping is taking the value in the rows list as an argument and return the string description.

**Comment by falkoschindler** [comment]
Sure. Have you seen https://nicegui.io/documentation/table#custom_sorting_and_formatting?

**Reply by laserir** [reply]
I have, but couldnt figure out how to call a function to manipulate the value within the `:format:`

**Reply by falkoschindler** [reply]
Something like this?
```py
def change_format():
    columns[1][':format'] = 'value => value + " years old"'
    table.update()

ui.button('Change format', on_click=change_format)
```

**Reply by laserir** [reply, marked as answer]
This works fine, but I want to send the value to a function, which returns the new value, e.g. like:
```py
def newValue(value):
    return value+1
```
`{'name': 'level', 'label': 'Level', 'field': 'level', ':format:': 'value => newValue(value)'},`
e.g. if the value is 1, the table would contain 2.

Anyhow, I solved it a little diffently - for anyone with a similar problem (simplyfied for better understanding of principle):
```py
columns = [
                {'name': 'system_id', 'label': 'ID', 'field': 'system_id'},
                {'name': 'level', 'label': 'Level', 'field': 'level'},
            ]

ui.table(columns=columns,rows=get_value(data))

level_mapping = {
    10: 'DEBUG',
    20: 'INFO',
    30: 'WARNING',
    40: 'ERROR',
    50: 'CRITICAL',
}

def get_value(input_data):
    return_data = []
    for system in input_data.get("system", []):
            return_system = {
                "system_id": system_id,
                "level": level_mapping.get(system.get("level")),
            }
            return_data.append(return_system)

    return return_data
```
